homeassistant/components/ozw/__init__.py

```diff
@@ -69,16 +69,46 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     data_nodes = {}
     data_values = {}
     removed_nodes = []
+    manager_options = {"topic_prefix": f"{TOPIC_OPENZWAVE}/"}
 
-    @callback
-    def send_message(topic, payload):
-        mqtt.async_publish(hass, topic, json.dumps(payload))
+    if entry.unique_id is None:
+        hass.config_entries.async_update_entry(entry, unique_id=DOMAIN)
+
+    if entry.data.get(CONF_USE_ADDON):
+        # Do not use MQTT integration. Use own MQTT client.
+        # Retrieve discovery info from the OpenZWave add-on.
+        discovery_info = await hass.components.hassio.async_get_addon_discovery_info(
+            "core_zwave"
+        )
 
-    options = OZWOptions(send_message=send_message, topic_prefix=f"{TOPIC_OPENZWAVE}/")
+        if not discovery_info:
+            _LOGGER.error("Failed to get add-on discovery info")
+            return False
```

Comment:
We should raise a ConfigEntryNotReady, to try again later. The add-on could be starting or could be started later by the user.

Reply:
Right, good point.